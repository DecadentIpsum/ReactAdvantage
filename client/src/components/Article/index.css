.article {
  margin-bottom: 15px; }

.article-top {
  display: flex;
  justify-content: space-between;
  align-items: center; }

.article-tag {
  color: #B3B3AB; }

.article-bottom {
  padding-left: 15px;
  padding-top: 15px;
  padding-bottom: 15px;
  border-left: 1px solid rgba(179, 179, 171, 0.7); }

.article-title {
  margin: 0;
  font-size: 1rem;
  color: rgba(30, 30, 30, 0.6); }

.article-text {
  font-size: 1rem;
  font-family: 'Roboto', sans-serif; }

.article-link {
  display: inline-block;
  font-size: 1rem;
  color: #5A88C3 !important;
  border-bottom: 1px dotted #5A88C3;
  text-decoration: none; }
  .article-link:hover {
    border-bottom-color: transparent; }

.tag {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 8px;
  background: #5A88C3;
  border-radius: 5px;
  color: #FFF; }
  .tag.enhancement {
    background: #edc02f; }

.tag-icon {
  margin-right: 5px; }

.tag-text {
  font-size: 1rem; }
