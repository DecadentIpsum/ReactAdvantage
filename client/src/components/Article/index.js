import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import Tag from 'components/Tag';
import './index.css';

export default class Article extends Component {
    render() {
        const {
            article: {
                date = '10th may, 2018',
                title = 'New and improved teamwork desc and teamwork projects integration',
                text = 'Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.',
                url = '/',
                type = this.props.index % 2,
            } = {}
        } = this.props;

        const tagText = type
            ? 'Enhancement'
            : 'Announcement';

        return (
            <article className="article">
                <div className="article-top">
                    <Tag
                        text={tagText}
                        type={tagText.toLowerCase()}
                    />
                    <div className="article-date">{`${date}`}</div>
                </div>

                <div className="article-bottom">
                    <h4 className="article-title">{title}</h4>
                    <p className="article-text">{text}</p>
                    <Link className="article-link" to={url}>
                        Learn more
                    </Link>
                </div>
            </article>
        );
    }
}
