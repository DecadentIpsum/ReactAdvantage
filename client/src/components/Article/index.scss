.article {
    margin-bottom: 15px;
}

.article-top {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.article-tag {
    color: #B3B3AB;
}

.article-bottom {
    padding-left: 15px;
    padding-top: 15px;
    padding-bottom: 15px;
    border-left: 1px solid rgba(#B3B3AB, 0.7);
}

.article-title {
    margin: 0;
    font-size: 1rem;
    color: rgba(#1E1E1E, 0.6);
}

.article-text {
    font-size: 1rem;
    font-family: 'Roboto', sans-serif;
}

.article-link {
    display: inline-block;
    font-size: 1rem;
    color: #5A88C3 !important;
    border-bottom: 1px dotted #5A88C3;
    text-decoration: none;

    &:hover {
        border-bottom-color: transparent;
    }
}

.tag {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 8px;
    background: #5A88C3;
    border-radius: 5px;
    color: #FFF;

    &.enhancement {
        background: #edc02f;
    }
}

.tag-icon {
    margin-right: 5px;
}

.tag-text {
    font-size: 1rem;
}