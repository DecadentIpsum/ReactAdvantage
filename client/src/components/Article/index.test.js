import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Article from './index';

it('renders correctly', () => {
    const component = shallow(<Article />);

    expect(component).toMatchSnapshot();
});
