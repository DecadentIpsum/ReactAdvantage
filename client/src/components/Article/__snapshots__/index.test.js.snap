// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="article-top"
>
          <Tag
                    text="Announcement"
                    type="announcement"
          />
          <div
                    className="article-date"
          >
                    10th may, 2018
          </div>
</div>,
        <div
          className="article-bottom"
>
          <h4
                    className="article-title"
          >
                    New and improved teamwork desc and teamwork projects integration
          </h4>
          <p
                    className="article-text"
          >
                    Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.
          </p>
          <Link
                    className="article-link"
                    replace={false}
                    to="/"
          >
                    Learn more
          </Link>
</div>,
      ],
      "className": "article",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Tag
              text="Announcement"
              type="announcement"
/>,
            <div
              className="article-date"
>
              10th may, 2018
</div>,
          ],
          "className": "article-top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "text": "Announcement",
              "type": "announcement",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "10th may, 2018",
              "className": "article-date",
            },
            "ref": null,
            "rendered": "10th may, 2018",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="article-title"
>
              New and improved teamwork desc and teamwork projects integration
</h4>,
            <p
              className="article-text"
>
              Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.
</p>,
            <Link
              className="article-link"
              replace={false}
              to="/"
>
              Learn more
</Link>,
          ],
          "className": "article-bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New and improved teamwork desc and teamwork projects integration",
              "className": "article-title",
            },
            "ref": null,
            "rendered": "New and improved teamwork desc and teamwork projects integration",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.",
              "className": "article-text",
            },
            "ref": null,
            "rendered": "Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Learn more",
              "className": "article-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Learn more",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="article-top"
>
            <Tag
                        text="Announcement"
                        type="announcement"
            />
            <div
                        className="article-date"
            >
                        10th may, 2018
            </div>
</div>,
          <div
            className="article-bottom"
>
            <h4
                        className="article-title"
            >
                        New and improved teamwork desc and teamwork projects integration
            </h4>
            <p
                        className="article-text"
            >
                        Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.
            </p>
            <Link
                        className="article-link"
                        replace={false}
                        to="/"
            >
                        Learn more
            </Link>
</div>,
        ],
        "className": "article",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Tag
                text="Announcement"
                type="announcement"
/>,
              <div
                className="article-date"
>
                10th may, 2018
</div>,
            ],
            "className": "article-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "text": "Announcement",
                "type": "announcement",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "10th may, 2018",
                "className": "article-date",
              },
              "ref": null,
              "rendered": "10th may, 2018",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="article-title"
>
                New and improved teamwork desc and teamwork projects integration
</h4>,
              <p
                className="article-text"
>
                Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.
</p>,
              <Link
                className="article-link"
                replace={false}
                to="/"
>
                Learn more
</Link>,
            ],
            "className": "article-bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New and improved teamwork desc and teamwork projects integration",
                "className": "article-title",
              },
              "ref": null,
              "rendered": "New and improved teamwork desc and teamwork projects integration",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.",
                "className": "article-text",
              },
              "ref": null,
              "rendered": "Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam eaque ipsa, quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt, explicabo. Nemo enim ipsam voluptatem, quia voluptas sit, aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos, qui ratione voluptatem sequi nesciunt, neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt, ut labore et dolore magnam aliquam quaerat voluptatem.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Learn more",
                "className": "article-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Learn more",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
